### PR TITLE
Validate and refactor SP-Responsive-UI.js 

### PR DESCRIPTION
**Pull request by populationtwo**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| New sample? | no |
| Related issues? | SP-Responsive-UI.js |
#### What's in this Pull Request?

Hi,

This PR is mainly code improvements. It's a copy of another PR I made on PnP-Tools https://github.com/OfficeDev/PnP-Tools/pull/79

### CODE REVIEW

**Comment by VesaJuvonen** [comment]
Hi Edwin,
thanks for the submission. Could you re-target this to Dev, so that we can get it processed. We only do master merge on monthly basis under the PnP core component repository. Thx.

**Comment by populationtwo** [comment]
@VesaJuvonen I have updated the PR, along with this [PR](https://github.com/OfficeDev/PnP-Tools/pull/79)

Thanks

File: Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js [hunk 1]
```diff
@@ -1,117 +1,121 @@
...
+    PnPResponsiveApp.init = function () {
+        if (!window.jQuery) {
+            // jQuery is needed for PnP Responsive UI to run, and is not fully loaded yet, try later
+            setTimeout(PnPResponsiveApp.init, 100);
+        } else {
+            $(function () { // only execute when DOM is fully loaded
+
+                // embedding and loading of all necessary CSS files and JS libraries
+                var currentScriptUrl = $('#PnPResponsiveUI').attr('src');
+                if (currentScriptUrl != "undefined") {
```
Comment (PaoloPia) [inline Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js]: Hi Edwin,
There is still one issue in this line. Could you please fix it before we can evaluate and eventually merge this PR?
Thanks.

**Comment by PaoloPia** [comment]
Hi Edwin,
We're evaluating your PR, as you saw in my previous comment. However, we would like to understand better the overall goal. In the PR description, you refer to a bug that you fixed. Can you be more specific about what you fixed and where?
Thanks.

File: Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js (hunk 1 above)
Comment (populationtwo) [inline Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js]: Hi @PaoloPia 

I'm sorry, what's the issue with that line? It was `undefined` but then as suggested on this PR, I changed it to `"undefined"`.

Thanks

Comment (VesaJuvonen) [inline Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js]: Hi Edwin, 
could you please come back on this or we'll need to close the PR. Thx.

**Comment by VesaJuvonen** [comment]
Hi Edwin,
just to follow up on this. We did check the implementation and it looks pretty nice, but we could not really see what is that exactly fixing or what was the actual change from behavior perspective. Could you provide more details on what does this fixes compared to previous version other than changes in the implementation style in CSS / JS files. thx for the additional details advance.

**Comment by populationtwo** [comment]
@PaoloPia My mistake. There was no bug, it was mainly code improvements. Fixed some issues I found on JSHint and refactored sideNavToggle.

Since this is an exact duplicate of my PR [here](https://github.com/OfficeDev/PnP-Tools/pull/79), should I copy the PR description from that page? I'm not sure if I should put the same description on both PR.

@VesaJuvonen 
There is no change from behaviors perspective. I basically run the SP-Responsive-UI.js through JShint to detect errors and potential problems. Let me know if you'd like to see a more detailed description from what I have on [here](https://github.com/OfficeDev/PnP-Tools/pull/79).

Thanks!

**Comment by PaoloPia** [comment]
Hi Edwin,
We really appreciate your contribution and your efforts. I manually merged some of your proposed changes in order to make proper testing. You can see the PR here (https://github.com/SharePoint/PnP-Sites-Core/pull/857). Thus, I will close this one, but you will be credited in the upcoming monthly community call. Thanks.

**Pull request closed without merging**